src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -862,6 +860,8 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
                 }
             }
 
+            factoryContext.Metadata.Add(new AcceptsMetadata(new string[] { "application/json" }));
```

Comment:
We should use a cached array. This is going to create an array per request delegate which is unnecessary. I imagine we could go further and cache the actual metadata.

```C#
private static readonly AcceptMetadata DefaultAcceptsMetadata = new(new[] { "application/json" });

....

factoryContext.Metadata.Add(DefaultAcceptMetadata);
```